File: Common/Events/OrderCreated.cs
using Common.Models;
using Common.Models.Payment;

namespace Common.Events
{
    public record OrderCreated
    {
        public Guid OrderId { get; init; }

        public required CustomerInfo CustomerInfo { get; init; }

        public required OrderStatus OrderStatus { get; init; }

        public required Address ShippingAddress { get; init; }

        public required PaymentInfo PaymentInfo { get; init; }

        public required IList<OrderItem> Items { get; init; } = new List<OrderItem>();

        public DateTime CreationDateTime { get; init; }
    }
}

File: rorderapi/src/events/ordercreated.rs
use chrono::{DateTime, Utc};
use serde::{Deserialize, Serialize};

use crate::models::{address::Address, customerinfo::CustomerInfo, orderitem::OrderItem, orderstatus::OrderStatus, paymentinfo::PaymentInfo};

use super::serializers::serialize_datetime;

#[derive(Serialize, Deserialize, Clone, Debug)]
#[serde(rename_all = "PascalCase")]
pub struct OrderCreated {
    pub order_id: bson::Uuid,
    pub customer_info: CustomerInfo,
    pub order_status: OrderStatus,
    pub shipping_address: Address,
    pub payment_info: PaymentInfo,
    pub items: Vec<OrderItem>,
    #[serde(serialize_with = "serialize_datetime")]
    pub creation_date_time: DateTime<Utc>
}